### DISCUSSION TITLE
TPC-H q9.sql performance lags behind xx, this test query profile, we can analyze it

### DISCUSSION THREAD

**Discussion by LuGuangming** [apache/doris#16675] [A - General / Q&A]
[tpch-q9SQL-profile.zip](https://github.com/apache/doris/files/10722248/tpch-q9SQL-profile.zip)

**Comment by LuGuangming** [comment]
[SR_q9_profile.zip](https://github.com/apache/doris/files/10728675/SR_q9_profile.zip)

**Comment by wangbo** [comment]
If you disable SR's pipeline switch, how much time may SR cost?
Currently Doris's pipeline query is under development.

**Reply by LuGuangming** [reply]
So let me test the result, when disable pipline, do you need to look at the profile that disable the pipline?

**Reply by LuGuangming** [reply]
![image](https://user-images.githubusercontent.com/71873108/221527034-4d1b3101-de88-4f32-90d8-62dcaf6aac18.png)
SR default pipeline is enable

**Comment by LuGuangming** [comment]
Doris q9.sql diff
Doris-1.2.0:   1m47s      
SR-2.4.2:       36s

**Reply by wangbo** [reply]
Can you show the result of ``` show variables``` for both SR and Doris?

**Reply by LuGuangming** [reply]
[doris_variables.txt](https://github.com/apache/doris/files/10838163/doris_variables.txt)
[sr_variables.txt](https://github.com/apache/doris/files/10838165/sr_variables.txt)

**Reply by wangbo** [reply]
Could you please ```set enable_pipeline_engine = false``` for SR and then re-run q9 to see time cost?

**Reply by LuGuangming** [reply]
[sr_q9_disable_pipeline.zip](https://github.com/apache/doris/files/10839879/sr_q9_disable_pipeline.zip)
SR disable pipeline cost time: 1m28s , We can find that the pipeline does improve performance
